diff --git a/client.c b/client.c
index afbf874..556efb4 100644
--- a/client.c
+++ b/client.c
@@ -93,7 +93,8 @@ client_init(Window win, struct screen_ctx *sc)
 	client_transient(cc);
 	client_mwm_hints(cc);
 
-	if ((cc->flags & CLIENT_IGNORE))
+	if ((cc->flags & CLIENT_IGNORE) ||
+	    (cc->flags & CLIENT_MAXFLAGS) == CLIENT_MAXIMIZED)
 		cc->bwidth = 0;
 	cc->dim.w = (cc->geom.w - cc->hint.basew) / cc->hint.incw;
 	cc->dim.h = (cc->geom.h - cc->hint.baseh) / cc->hint.inch;
@@ -308,6 +309,8 @@ client_toggle_sticky(struct client_ctx *cc)
 void
 client_toggle_fullscreen(struct client_ctx *cc)
 {
+	if (cc->flags & CLIENT_IGNORE)
+		return;
 	struct screen_ctx	*sc = cc->sc;
 	struct geom		 area;
 
@@ -316,8 +319,9 @@ client_toggle_fullscreen(struct client_ctx *cc)
 		return;
 
 	if (cc->flags & CLIENT_FULLSCREEN) {
-		if (!(cc->flags & CLIENT_IGNORE))
-			cc->bwidth = Conf.bwidth;
+		if (!(cc->flags & CLIENT_IGNORE) &&
+			!((cc->flags & CLIENT_MAXFLAGS) == CLIENT_MAXIMIZED))
+				cc->bwidth = Conf.bwidth;
 		cc->geom = cc->fullgeom;
 		cc->flags &= ~(CLIENT_FULLSCREEN | CLIENT_FREEZE);
 		goto resize;
@@ -342,6 +346,8 @@ resize:
 void
 client_toggle_maximize(struct client_ctx *cc)
 {
+	if (cc->flags & CLIENT_IGNORE)
+		return;
 	struct screen_ctx	*sc = cc->sc;
 	struct geom		 area;
 
@@ -349,6 +355,7 @@ client_toggle_maximize(struct client_ctx *cc)
 		return;
 
 	if ((cc->flags & CLIENT_MAXFLAGS) == CLIENT_MAXIMIZED) {
+		cc->bwidth = Conf.bwidth;
 		cc->geom = cc->savegeom;
 		cc->flags &= ~CLIENT_MAXIMIZED;
 		goto resize;
@@ -368,6 +375,7 @@ client_toggle_maximize(struct client_ctx *cc)
 	    cc->geom.x + cc->geom.w / 2,
 	    cc->geom.y + cc->geom.h / 2, 1);
 
+	cc->bwidth = 0;
 	cc->geom.x = area.x;
 	cc->geom.y = area.y;
 	cc->geom.w = area.w - (cc->bwidth * 2);
@@ -383,6 +391,8 @@ resize:
 void
 client_toggle_vmaximize(struct client_ctx *cc)
 {
+	if (cc->flags & CLIENT_IGNORE)
+		return;
 	struct screen_ctx	*sc = cc->sc;
 	struct geom		 area;
 
@@ -390,6 +400,7 @@ client_toggle_vmaximize(struct client_ctx *cc)
 		return;
 
 	if (cc->flags & CLIENT_VMAXIMIZED) {
+		cc->bwidth = Conf.bwidth;
 		cc->geom.y = cc->savegeom.y;
 		cc->geom.h = cc->savegeom.h;
 		cc->flags &= ~CLIENT_VMAXIMIZED;
@@ -416,6 +427,8 @@ resize:
 void
 client_toggle_hmaximize(struct client_ctx *cc)
 {
+	if (cc->flags & CLIENT_IGNORE)
+		return;
 	struct screen_ctx	*sc = cc->sc;
 	struct geom		 area;
 
@@ -423,6 +436,7 @@ client_toggle_hmaximize(struct client_ctx *cc)
 		return;
 
 	if (cc->flags & CLIENT_HMAXIMIZED) {
+		cc->bwidth = Conf.bwidth;
 		cc->geom.x = cc->savegeom.x;
 		cc->geom.w = cc->savegeom.w;
 		cc->flags &= ~CLIENT_HMAXIMIZED;
@@ -449,12 +463,16 @@ resize:
 void
 client_resize(struct client_ctx *cc, int reset)
 {
+	if (cc->flags & CLIENT_IGNORE)
+		return;
 	if (reset) {
+		cc->bwidth = Conf.bwidth;
 		cc->flags &= ~CLIENT_MAXIMIZED;
 		xu_ewmh_set_net_wm_state(cc);
 	}
 
-	client_draw_border(cc);
+	if ((cc->flags & CLIENT_MAXFLAGS) != CLIENT_MAXIMIZED)
+		client_draw_border(cc);
 
 	XMoveResizeWindow(X_Dpy, cc->win, cc->geom.x,
 	    cc->geom.y, cc->geom.w, cc->geom.h);
@@ -466,6 +484,8 @@ client_resize(struct client_ctx *cc, int reset)
 void
 client_move(struct client_ctx *cc)
 {
+	if (cc->flags & CLIENT_IGNORE)
+		return;
 	XMoveWindow(X_Dpy, cc->win, cc->geom.x, cc->geom.y);
 	client_config(cc);
 }
@@ -544,6 +564,8 @@ client_ptr_save(struct client_ctx *cc)
 void
 client_hide(struct client_ctx *cc)
 {
+	if (cc->flags & CLIENT_IGNORE)
+		return;
 	XUnmapWindow(X_Dpy, cc->win);
 
 	if (cc->flags & CLIENT_ACTIVE) {
@@ -652,6 +674,8 @@ client_wm_hints(struct client_ctx *cc)
 void
 client_close(struct client_ctx *cc)
 {
+	if (cc->flags & CLIENT_IGNORE)
+		return;
 	if (cc->flags & CLIENT_WM_DELETE_WINDOW)
 		xu_send_clientmsg(cc->win, cwmh[WM_DELETE_WINDOW], CurrentTime);
 	else
@@ -869,7 +893,7 @@ client_transient(struct client_ctx *cc)
 	if (XGetTransientForHint(X_Dpy, cc->win, &trans)) {
 		if ((tc = client_find(trans)) != NULL) {
 			if (tc->flags & CLIENT_IGNORE) {
-				cc->flags |= CLIENT_IGNORE;
+				cc->flags |= CLIENT_IGNORE | CLIENT_FREEZE;
 				cc->bwidth = tc->bwidth;
 			}
 		}
@@ -938,9 +962,11 @@ client_htile(struct client_ctx *cc)
 	if (n == 0)
 		return;
 
+	if (cc->flags & CLIENT_FULLSCREEN)
+		return;
 	if (cc->flags & CLIENT_VMAXIMIZED ||
 	    cc->geom.h + (cc->bwidth * 2) >= area.h)
-		return;
+		cc->bwidth = Conf.bwidth;
 
 	cc->flags &= ~CLIENT_HMAXIMIZED;
 	cc->geom.x = area.x;
@@ -1007,9 +1033,11 @@ client_vtile(struct client_ctx *cc)
 	if (n == 0)
 		return;
 
+	if (cc->flags & CLIENT_FULLSCREEN)
+		return;
 	if (cc->flags & CLIENT_HMAXIMIZED ||
 	    cc->geom.w + (cc->bwidth * 2) >= area.w)
-		return;
+		cc->bwidth = Conf.bwidth;
 
 	cc->flags &= ~CLIENT_VMAXIMIZED;
 	cc->geom.x = area.x;
diff --git a/conf.c b/conf.c
index 5b11f67..1ea9c84 100644
--- a/conf.c
+++ b/conf.c
@@ -472,7 +472,7 @@ conf_client(struct client_ctx *cc)
 
 	TAILQ_FOREACH(wn, &Conf.ignoreq, entry) {
 		if (strncasecmp(wn->name, cc->name, strlen(wn->name)) == 0) {
-			cc->flags |= CLIENT_IGNORE;
+			cc->flags |= CLIENT_IGNORE | CLIENT_FREEZE;
 			break;
 		}
 	}
